src/applications/disability-benefits/all-claims/tests/components/ReviewCardField.unit.spec.jsx

```diff
@@ -56,12 +56,14 @@ describe('Schemaform: ReviewCardField', () => {
 
   it('should throw an error if no viewComponent is found', () => {
     expect(() => {
+      // eslint-disable-next-line enzyme-always-unmount/no-matching-unmount
       shallow(<ReviewCardField {...defaultProps} uiSchema={{}} />);
```

Comment:
:thinking: Should we ignore this? That is, will this potentially put our testing in the same weird state that we're trying to prevent with this plugin?

If not, we can avoid having to throw an ignore line in here by adding a `VariableDeclaration` to the AST selector so it ignores these calls to `shallow` and `mount` if they're not assigned to a variable.

Reply:
If it is possible to still get into whatever weird state we're trying to prevent, could we tack on a `.unmount()`?
```js
expect(() => {
  shallow(...).unmount();
}).to.throw(...);
```

Reply:
I think the linter should catch when components are mounted but not assigned. It's not possible to do in this component because it throws an error in the constructor- enzyme only returns an instance _after_ the constructor and render are called. 

I think it's weird that this component props validation in the `constructor`. I would rather refactor the component because
- props validation should happen on every render- (putting it in the constructor only validates the props when the component is initialized), and
- components should render an error state and leave error handling to non-React js.

Reply:
The second point is debatable- we don't really have any good directives on error handling. Might do an RFC for this later- I think it's cleaner to separate error handling concern from view rendering and to make views as dumb as possible.

Reply:
We're unmounting components as a defensive measure. I think in the case it's safe to ignore because- there's no async behavior in this component that might call js later. so there should be no reference to this component after this unit test is complete. 

Ideally we don't ignore and let the linter work but I figured we could refactor the component later. We're doing this blanket `unmount` ing to make it easier to reason though- also to catch problems by async behavior that might be added during maintenance. Since these bugs are so hard to track down, I figured it would be best just to enforce unmounting always even if it's not necessary (it's recommended by enzyme maintainers).

Reply:
Yeah, we could probably make better use of PropTypes. I had this idea that throwing an error in the constructor with a very specific message when the developer clearly messed up the `formConfig` would be more helpful than just relying on PropTypes. Now that I think about it again, I think they'd accomplish the same thing.

That said, there may be more of that kind of validation that makes sense to happen in the constructor and we should account for that. If the prop comes only from being hard coded, not any user input, and had validation that PropTypes won't catch...

It's pretty niche, though. PropTypes validations can be pretty thorough. :man_shrugging:

Reply:
We could add this behavior to a static method so it doesn't need to be instantiated / mounted to be unit tested. Then you could have two unit tests:

- on that mocks the method to validate it's called from the `constructor` and
- one that calls the static method on the definition to verify that it throws an error.

Reply:
:thinking: Probably getting into the weeds a bit here, but would this be just so testing is easier?

Reply:
Yes- I also think it's good design to just call methods/functions from life cycle methods instead of having logic in them. Static methods are a good place for helper functions conceptually since they don't make any changes to the instance.

Reply:
Cool.

Anywho, I think I agree with you. We can ignore these lines and refactor the component later. If we end up writing components where we have to add more ignore lines, it's probably a sign that the component should be written differently.